services/cuckoo/processing.go

```diff
@@ -0,0 +1,234 @@
+	if err == nil && mStruct != nil {
+		machineString = mStruct.Name
+	}
+
+	resMap := make(map[string]interface{})
+	resMap["started"] = i.Started
+	resMap["ended"] = i.Ended
+	resMap["analysis_id"] = strconv.Itoa(i.Id)
+
+	return []*CrtResult{&CrtResult{"info", machineString, resMap}}
+}
+
+// processReportSignatures extracts all the data from the signatures
+// section of the cuckoo report struct.
+func processReportSignatures(sigs []*cuckoo.TasksReportSignature) []*CrtResult {
+	if sigs == nil {
+		return []*CrtResult{}
+	}
+
+	l := len(sigs)
+	res := make([]*CrtResult, l, l)
+	resMap := make(map[string]interface{})
+
+	for k, sig := range sigs {
+		resMap["severity"] = strconv.Itoa(sig.Severity)
+		resMap["name"] = sig.Name
+
+		res[k] = &CrtResult{
+			"signature",
+			sig.Description,
+			resMap,
+		}
+	}
+
+	return res
+}
+
+// processReportBehavior extracts all the data from the behavior
+// section of the cuckoo report struct.
+func processReportBehavior(behavior *cuckoo.TasksReportBehavior) []*CrtResult {
+	if behavior == nil {
+		return []*CrtResult{}
+	}
+
+	var res []*CrtResult
+	resMap := make(map[string]interface{})
+
+	if behavior.Processes != nil {
+		for _, p := range behavior.Processes {
+			resMap["process_id"] = strconv.Itoa(p.Id)
+			resMap["parent_id"] = strconv.Itoa(p.ParentId)
+			resMap["first_seen"] = p.FirstSeen
+
+			res = append(res, &CrtResult{
+				"process",
+				p.Name,
+				resMap,
+			})
+		}
+	}
+
+	// push api calls
+	// not mixed in with upper loop so we can make it optional later
+	if behavior.Processes != nil {
+		pushCounter := 0
+
+		for _, p := range behavior.Processes {
+
+			procDescription := fmt.Sprintf("%s (%d)", p.Name, p.Id)
+			for _, c := range p.Calls {
+
+				if pushCounter >= ctx.Config.Settings.MaxAPICalls {
+					break
+				}
+
+				resMap := make(map[string]interface{})
+				resMap["category"] = c.Category
+				resMap["status"] = c.Status
+				resMap["return"] = c.Return
+				resMap["timestamp"] = c.Timestamp
+				resMap["thread_id"] = c.ThreadId
+				resMap["repeated"] = c.Repeated
+				resMap["api"] = c.Api
+				resMap["id"] = c.Id
+				resMap["process"] = procDescription
+				resMap["arguments"] = c.Arguments
+
+				res = append(res, &CrtResult{
+					"api_call",
+					c.Api,
+					resMap,
+				})
+				pushCounter += 1
+			}
+		}
+	}
+
+	if behavior.Summary != nil {
+		if behavior.Summary.Files != nil {
+			for _, b := range behavior.Summary.Files {
+				res = append(res, &CrtResult{
+					"file",
+					b,
+					nil,
+				})
+			}
+		}
+
+		if behavior.Summary.Keys != nil {
+			for _, b := range behavior.Summary.Keys {
+				res = append(res, &CrtResult{
+					"registry_key",
+					b,
+					nil,
+				})
+			}
+		}
+
+		if behavior.Summary.Mutexes != nil {
+			for _, b := range behavior.Summary.Mutexes {
+				res = append(res, &CrtResult{
+					"mutex",
+					b,
+					nil,
+				})
+			}
+		}
+	}
+
+	return res
+}
+
+/*
+// support for dropped files will be added later
```

Comment:
Should maybe be a special flag with a huge warning about this. This can bloat your system really fast.